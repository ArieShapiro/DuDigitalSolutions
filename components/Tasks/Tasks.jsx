import React, { PureComponent } from 'react';
import Task from '../Task/Task';

class Tasks extends PureComponent {
	render () {
		const { tasks, toggleTask } = this.props;

		return (
			<div className="tasks">
				{
					tasks.map((task) => {
						return (
							<div className="task-item" key={`task-${task.id}`}>
								<Task
									task={task}
									toggleTask={toggleTask}
								/>
							</div>
						);
					})
				}
			</div>
		);
	}
}

export default Tasks;
