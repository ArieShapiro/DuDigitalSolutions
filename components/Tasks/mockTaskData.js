const mockTaskData = [];

mockTaskData.push({
	id: 1,
	done: false,
	title: 'Center the "brand-box" in our header',
	description: 'Please make sure the "brand-box" is perfectly centered vertically and horizontally even if we start changing the height of ".header".\nYou get free cookies if your solution doesn\'t depend on hardcoded px heights at all. ;)',
	reason: 'We do a lot of complex coding, but CSS is still the basis of every beautiful web application. ;)',
	estimatedTime: '2-3 Min',
});

mockTaskData.push({
	id: 2,
	done: false,
	title: 'Make the "delete" button work',
	description: 'At the bottom right of every task item you have a [delete] button, that does nothing right now.\nPlease go ahead and make it work by adding a redux action and manipulating the store to remove the corresponding task on click.',
	reason: 'We want to see working with Redux basics. Although we don\'t always use Redux, it\'s still one of the most important flux implementations to know.',
	estimatedTime: '5-8 Min',
});

mockTaskData.push({
	id: 3,
	done: false,
	title: 'Describe your setup',
	description: 'Now a theoretical question: We want to know how you would tackle a big project.\nImagine you have to build a SPA of a complex application like a social media platform containing multiple entities (users, profiles, posts, comments, messages, ...).\n\n* What react related packages and setup would you choose?     (store management, type checking, codestyle, testing, ...) \n*How would you improve our simple example to make it more readable/cleaner?\n\nPlease add your answers in the mockTaskData.js file under "answer1" and "answer2".',
	reason: 'This will give us an idea which packages and technologies you already worked with\nand also provide a good basis for our interview',
	estimatedTime: '3-10 Min',
	answer1: "", // TODO: put your answer to the stack question here
	answer2: "", // TODO: put answer to improvements here
});

mockTaskData.push({
	id: 4,
	done: false,
	title: 'Fix the testcases to make them all pass',
	description: 'At the moment, the unit tests will fail, because of an error we worked in there for you. Only small adjustments are needed to make all the tests in this application pass.\n\n(This task is already advanced. If you can solve it: AWESOME. If not: don\'t worry about it. We want you to grow and learn with us and that\'s a good topic to get started with. ;))',
	reason: 'Please fix our broken application and show us how awesome you are at working with tests. ;)',
	estimatedTime: '5-15 Min',
});

mockTaskData.push({
	id: 5,
	done: false,
	title: 'Send everything over',
	description: 'Please zip the project folder, containing the code you\'ve changed, and send it to jobs@du.digital',
	reason: 'Thank you for taking part in our little test. We will get back to you as soon as we looked it over',
	estimatedTime: '1 Min',
});

export default mockTaskData;
