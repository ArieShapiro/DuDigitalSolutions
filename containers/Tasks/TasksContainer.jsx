import React, { Component } from 'react';
import { connect } from 'react-redux';
import Tasks from '../../components/Tasks/Tasks';
import { toggleTask } from '../../store';

class TasksContainer extends Component {
	render () {
		const { tasks, toggleTask } = this.props;

		return (
			<Tasks
				tasks={tasks}
				toggleTask={toggleTask}
			/>
		);
	}
}

function mapStateToProps (state) {
	const { tasks } = state;
	return { tasks };
}

function mapDispatchToProps (dispatch) {
	return {
		toggleTask: (taskId) => {
			dispatch(toggleTask(taskId));
		},
	};
}

export default connect(mapStateToProps, mapDispatchToProps)(TasksContainer);
