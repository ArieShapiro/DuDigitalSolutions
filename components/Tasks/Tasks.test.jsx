import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import Tasks from './Tasks';
import mockTaskData from './mockTaskData';

describe('task listing actions', () => {

	it('renders correctly', () => {
		const renderer = new ShallowRenderer();
		renderer.render(<Tasks tasks={mockTaskData}/>);

		const tree = renderer.getRenderOutput();
		expect(tree).toMatchSnapshot();
	});

	it('has the correct amount of task items', () => {
		const renderer = shallow(<Tasks tasks={mockTaskData}/>);
		expect(renderer.find('.task-item').length).toEqual(mockTaskData.length);
	});

});