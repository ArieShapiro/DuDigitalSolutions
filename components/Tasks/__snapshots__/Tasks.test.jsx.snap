// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`task listing actions renders correctly 1`] = `
<div
  className="tasks"
>
  <div
    className="task-item"
  >
    <Task
      task={
        Object {
          "description": "Please make sure the \\"brand-box\\" is perfectly centered vertically and horizontally even if we start changing the height of \\".header\\".
You get free cookies if your solution doesn't depend on hardcoded px heights at all. ;)",
          "done": false,
          "estimatedTime": "2-3 Min",
          "id": 1,
          "reason": "We do a lot of complex coding, but CSS is still the basis of every beautiful web application. ;)",
          "title": "Center the \\"brand-box\\" in our header",
        }
      }
    />
  </div>
  <div
    className="task-item"
  >
    <Task
      task={
        Object {
          "description": "At the bottom right of every task item you have a [delete] button, that does nothing right now.
Please go ahead and make it work by adding a redux action and manipulating the store to remove the corresponding task on click.",
          "done": false,
          "estimatedTime": "5-8 Min",
          "id": 2,
          "reason": "We want to see working with Redux basics. Although we don't always use Redux, it's still one of the most important flux implementations to know.",
          "title": "Make the \\"delete\\" button work",
        }
      }
    />
  </div>
  <div
    className="task-item"
  >
    <Task
      task={
        Object {
          "answer1": "",
          "answer2": "",
          "description": "Now a theoretical question: We want to know how you would tackle a big project.
Imagine you have to build a SPA of a complex application like a social media platform containing multiple entities (users, profiles, posts, comments, messages, ...).

* What react related packages and setup would you choose?     (store management, type checking, codestyle, testing, ...) 
*How would you improve our simple example to make it more readable/cleaner?

Please add your answers in the mockTaskData.js file under \\"answer1\\" and \\"answer2\\".",
          "done": false,
          "estimatedTime": "3-10 Min",
          "id": 3,
          "reason": "This will give us an idea which packages and technologies you already worked with
and also provide a good basis for our interview",
          "title": "Describe your setup",
        }
      }
    />
  </div>
  <div
    className="task-item"
  >
    <Task
      task={
        Object {
          "description": "At the moment, the unit tests will fail, because of an error we worked in there for you. Only small adjustments are needed to make all the tests in this application pass.

(This task is already advanced. If you can solve it: AWESOME. If not: don't worry about it. We want you to grow and learn with us and that's a good topic to get started with. ;))",
          "done": false,
          "estimatedTime": "5-15 Min",
          "id": 4,
          "reason": "Please fix our broken application and show us how awesome you are at working with tests. ;)",
          "title": "Fix the testcases to make them all pass",
        }
      }
    />
  </div>
  <div
    className="task-item"
  >
    <Task
      task={
        Object {
          "description": "Please zip the project folder, containing the code you've changed, and send it to jobs@du.digital",
          "done": false,
          "estimatedTime": "1 Min",
          "id": 5,
          "reason": "Thank you for taking part in our little test. We will get back to you as soon as we looked it over",
          "title": "Send everything over",
        }
      }
    />
  </div>
</div>
`;
